default.nix

```diff
@@ -0,0 +1,83 @@
+let
+  inputs = import ./flake-inputs.nix;
```

Comment:
Incidentally I was working on exactly that question yesterday: should inputs be an argument or hard-coded? I came to the conclusion that they should be an argument, because you may have arbitrarily complex incantations in the default values for other arguments, and may only want to change the underlying source files. (Somewhat related, it may be good practice to export inputs in the output attrset, because that would allow patching the default value by calling the function without arguments, picking and updating the inputs, and calling the function with the updated inputs - essentially an `.override`)

The weird bit to me is the object flake-compat produces: it's already something evaluated rather than something akin to a lockfile entry (as in npins). Not sure we should open that can of worms here though...

Reply:
I only allow `outPath` from `flake.defaultNix.inputs`, though. I guess I'll put `fetchTree` somewhere in the tree to replace flake-compat.

Reply:
It's a lot more complicated than I thought, I ended up forking flake-compat and modifying it so that it doesn't touch `flake.nix` at all.